src/packageManager/FileDownloader.ts

```diff
@@ -53,15 +54,13 @@ async function downloadFile(fd: number, description: string, urlString: string,
         rejectUnauthorized: util.isBoolean(strictSSL) ? strictSSL : true
     };
 
-    return new Promise<void>((resolve, reject) => {
-        if (fd == 0) {
-            reject(new NestedError("Temporary package file unavailable"));
-        }
+    let buffers: any[] = [];
 
+    return new Promise<Buffer>((resolve, reject) => {
         let request = https.request(options, response => {
             if (response.statusCode === 301 || response.statusCode === 302) {
                 // Redirect - download from new location
-                return resolve(downloadFile(fd, description, response.headers.location, eventStream, networkSettingsProvider));
+                return resolve(downloadFile(description, response.headers.location, eventStream, networkSettingsProvider));
```

Comment:
why on this invocation of `resolve` we return its return value, but on other invocations we ignore it?

Reply:
I am not sure about that. Do you think we need to change this?